PocketKit/Sources/PocketKit/Home/HomeViewModel.swift

```diff
@@ -44,391 +43,209 @@ class HomeViewModel {
+        if !recentSaves.isEmpty {
+            snapshot.appendSections([.recentSaves])
+            snapshot.appendItems(
+                recentSaves.map { .recentSaves($0.objectID) },
+                toSection: .recentSaves
+            )
         }
-    }
-        
-    private func _share(item: SavedItem, sender: Any?) {
-        sharedActivity = PocketItemActivity(url: item.url, sender: sender)
-    }
-    
-    private func _archive(item: SavedItem) {
-        source.archive(item: item)
-    }
 
-    private func confirmDelete(item: SavedItem) {
-        presentedAlert = PocketAlert(
-            title: "Are you sure you want to delete this item?",
-            message: nil,
-            preferredStyle: .alert,
-            actions: [
-                UIAlertAction(title: "No", style: .default) { [weak self] _ in
-                    self?.presentedAlert = nil
-                },
-                UIAlertAction(title: "Yes", style: .destructive) { [weak self] _ in
-                    self?.presentedAlert = nil
-                    self?._delete(item: item)
-                }
-            ],
-            preferredAction: nil
-        )
-    }
-    
-    private func _delete(item: SavedItem) {
-        presentedAlert = nil
-        source.delete(item: item)
-    }
+        if let slateLineup = slateLineup,
+           let slates = slateLineup.slates?.compactMap({ $0 as? Slate }) {
 
-    func reportAction(for cell: HomeViewModel.Cell, at indexPath: IndexPath) -> ItemAction? {
-        return .report { [weak self] _ in
-            self?.report(cell, at: indexPath)
-        }
-    }
+            for slate in slates {
+                guard var recs = slate.recommendations?.compactMap({ $0 as? Recommendation }),
+                      !recs.isEmpty else {
+                    continue
+                }
 
-    func saveAction(for cell: HomeViewModel.Cell, at indexPath: IndexPath) -> ItemAction? {
-        guard let objectID = getRecommendationID(with: cell),
-              let viewModel = viewModel(for: objectID) else {
-            return nil
-        }
+                let hero = recs.removeFirst()
+                snapshot.appendSections([.slateHero(slate.objectID)])
+                snapshot.appendItems(
+                    [.recommendationHero(hero.objectID)],
+                    toSection: .slateHero(slate.objectID)
+                )
 
-        return .recommendationPrimary { [weak self] _ in
-            if viewModel.isSaved {
-                self?.archive(cell, at: indexPath)
-            } else {
-                self?.save(cell, at: indexPath)
+                guard !recs.isEmpty else {
+                    continue
+                }
+                snapshot.appendSections([.slateCarousel(slate.objectID)])
+                snapshot.appendItems(
+                    recs[0...min(3, recs.endIndex - 1)].map { .recommendationCarousel($0.objectID) },
+                    toSection: .slateCarousel(slate.objectID)
+                )
```

Comment:
previously, we didn't limit it to 3, why not just rely on passing the right data and use `recs.endIndex - 1`?

Reply:
So we're accounting for this scenario:

1. User launches app and lands on Home
2. We fetch the entire slate lineup, with a maximum number of recommendations per slate of 5
3. User taps "See All" for a given slate
4. We fetch up to 25 recommendations for the slate the user tapped on
5. User pops back to Home
6. Now we have 25 recommendations in the database but we don't want that many in the carousel

We were previously accounting for this by using some more brute force logic to delete all the extra recommendations when popping back to home (there was a line in `CompactHomeCoordinator` that did this). But I thought it made more sense to leave the recommendations in the database until a full refresh of the slate lineup (that way we show all the recommendations if the user taps "See All" again).

Reply:
got it thanks @jacobsimeon  :) This is the function you are referring to right? `resetSlate(keeping: 5)`